// background-image-on-load/src/App.js
import React, { Component } from 'react';
import BackgroundImageOnLoad from 'background-image-on-load';
import logo from './logo.svg';
import './App.css';

class App extends Component {

  state = {
    bgIsLoaded: false
  };
  
  render() {
    const { bgIsLoaded } = this.state;
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Background-Image-On-Load</h1>
        </header>
        <div
          style={{ 
            height: 300,
            backgroundPosition: 'center',
            backgroundSize: 'cover',
            backgroundImage: `url(${!bgIsLoaded ? 'https://picsum.photos/310/310?blur' : 'https://picsum.photos/1600/310'})`
          }}
        >
          <BackgroundImageOnLoad
            src={'https://unsplash.it/1200/310?random'}
            onLoadBg={() => 
              this.setState({
              bgIsLoaded: true
            })}
          />
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
